src/Microsoft.AspNet.Mvc.Core/Resources.resx

```diff
@@ -319,4 +319,7 @@
     <value>Path '{0}' was not rooted.</value>
     <comment>{0} is the path which wasn't rooted</comment>
   </data>
+  <data name="DefaultContentTypeMustHaveEncoding" xml:space="preserve">
+    <value>The default content type '{0}' must have an encoding.</value>
```

Comment:
Your feedback about the names is :100: correct. However we shouldn't even be throwing exceptions here. This is internal code.